plugins/xpressive/expressive_plugin.cpp

```diff
@@ -0,0 +1,926 @@
+/*
+ * expressive_plugin.cpp - instrument which uses a mathematical formula parser
+ *
+ * Copyright (c) 2016-2017 Orr Dvori
+ * 
+ * This file is part of LMMS - http://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "expressive_plugin.h"
+
+#include <QDomElement>
+//#include <QDebug>
+#include "SubWindow.h"
+#include "GuiApplication.h"
+#include "MainWindow.h"
+#include "exprfront.h"
+
+#include "base64.h"
+#include "Engine.h"
+#include "Graph.h"
+#include "InstrumentTrack.h"
+#include "Knob.h"
+#include "LedCheckbox.h"
+#include "Mixer.h"
+#include "NotePlayHandle.h"
+#include "Oscillator.h"
+#include "PixmapButton.h"
+#include "templates.h"
+#include "ToolTip.h"
+#include "Song.h"
+#include "lmms_constants.h"
+//#include "interpolation.h"
+
+#include "embed.cpp"
+
+extern "C" {
+
+Plugin::Descriptor PLUGIN_EXPORT xpressive_plugin_descriptor = { STRINGIFY(
+		PLUGIN_NAME), "X-Pressive", QT_TRANSLATE_NOOP("pluginBrowser",
+		"Mathematical expression parser"), "Orr Dvori", 0x0100,
+		Plugin::Instrument, new PluginPixmapLoader("logo"), NULL, NULL };
+
+}
+
+exprSynth::exprSynth(const WaveSample *gW1, const WaveSample *gW2, const WaveSample *gW3, ExprFront *_exprO1, ExprFront *_exprO2,
+					NotePlayHandle *_nph, const sample_rate_t _sample_rate, const FloatModel* _pan1, const FloatModel* _pan2, float _rel_trans)
+	:exprO1(_exprO1),exprO2(_exprO2),W1(gW1),W2(gW2),W3(gW3),nph(_nph),sample_rate(_sample_rate),pan1(_pan1),pan2(_pan2),rel_transition(_rel_trans)
```

Comment:
All kind of types begin with an upper case letter. Member variables are prefixed with `m_`.

Although this is work in progress, please follow the format style (one line per member) used in other parts of LMMS. Other developers may need to fix your code or help you.